lrlex/src/lib/parser.rs

```diff
@@ -14,7 +14,6 @@ lazy_static! {
         Regex::new(r"^%[sS][a-zA-Z0-9]*$").unwrap();
     static ref RE_EXCLUSIVE_START_STATE_DECLARATION: Regex =
         Regex::new(r"^%[xX][a-zA-Z0-9]*$").unwrap();
-    static ref RE_ERE_QUOTED_CHARS: Regex = Regex::new(r#"([\^|\.|\[\$|\(|\)|\*|\+|\?])"#).unwrap();
     static ref RE_LEX_ESC: Regex =
         Regex::new(r#"([:xdigit:]|[digit]|[\\|a|b|f|n|r|t|v])"#).unwrap();
```

Comment:
So I think that this needs some work, In particular the `xdigit`and `digit` should if I'm understanding the lex spec should
be matching words rather than characters, so I don't think this is right yet.

Or at least I still need to look into this better.

Reply:
> \xdigits A <backslash> character followed by the longest sequence of hexadecimal-digit characters
> (01234567abcdefABCDEF). If all of the digits are 0 (that is, representation of the NUL character), the behavior is undefined.

So I guess part of my confusion with this regex is my reading of the above says this should be:
`\a' should be interpreted as a ascii bell and as a `\a` xdigit.

I think it is just not mentioning `\xa` is hex and `\a` is bell.  I.e. I think it should be "a <backslash> followed by an 'x' followed by ...', alas I haven't really used hex escaping in lex and haven't found any examples of it that might clarify things.

Reply:
So, the only other thing i've found so far is:
> \xDigits The character whose encoding is represented by the sequence of hexadecimal characters specified by the Digits string. 

from https://www.ibm.com/docs/en/aix/7.1?topic=information-extended-regular-expressions-in-lex-command

So I am thinking that this is just the *normal* `\xf00`, and not some ancient lex mishap implying that that regex should be `x[:xdigits:]+`

I have changed this, in https://github.com/softdevteam/grmtools/pull/333/commits/6020226bd8fa32a57554e6e4a5666f20f667659e In doing so I'm convinced that we can remove some TODO I had added in previous commits.